### ISSUE TITLE
Movespeed 2DA randomization results in softlocks when affecting story-critical custcene characters

### ISSUE THREAD

**Issue by CocaineJohnsson** [LaneDibello/Kotor-Randomizer#107]
For example what I ran into on my current seed: 

The unavoidable encounter with Commander Grann (Sith Embassy/Manaan), his movement speed is 0 due to movement speed rando and he will therefore never successfully leave the elevator, this softlocks the cutscene (and by extension the rest of the game since this is a mandatory part of progression).

There are a few obvious solutions, as far as I see it anyway. A few of the simpler solutions:

1. Disallow a movespeed of 0 entirely
2. Modify the cutscenes so that if an NPC takes too long to reach their trigger we just proceed anyway or teleport them to the trigger
3. Flag certain NPCs as disallowing a movespeed of 0 (and probably flag things like turrets to only allow a movespeed of 0)
4. Allow the user to select a movespeed range and warn them if 0 is an option.

There are other solutions as well, but those are the obvious and trivial ones.

[Spoiler log](https://github.com/LaneDibello/Kotor-Randomizer/files/12065362/2023-07-14.08-27-36.Seed.365937999.xlsx)

**Comment by LaneDibello** [comment]
Yeah we've encountered this before... Usually we've advised others to avoid movement speed randomization for this reason. 

Your first solution is the simplest patch by far, and would be as simple as adding a special case for this 2DA that omits the stationary movement speed. 

Modifying cut scenes like this wouldn't be feasible programmatically, though I do know how I could do it by hand. However, such a patch would be large and time consuming.

Similarly, preventing NPCs from gaining this adjusted movement speed would not only require us to crawl every single NPC, in every RIM and Template (runtime consuming) but also to keep track of what the adjusted movement speeds map to.

The movespeed range selection isn't actually a bad idea. It could be quite fun to let the user just set some bounds and we'll randomly grab numbers between there. The only draw back being is this would be more than just a special case for this particular randomization category, and would instead warrant an additional feature (not out of the question, especially in our future plans for 2DAs).

Unfortunately, neither Glasnonck, nor myself are actively maintaining this project. Though, we do plan to return to it at some point. Feel free to branch and mess with it yourself in the meantime. And I'll be happy to answer any questions in the interim.
